lib/hexo/validate_config.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+module.exports = ctx => {
+  const { config, log } = ctx;
+
+  log.info('Validating config');
+
+  // Validation for config.url && config.root
+  if (typeof config.url !== 'string') {
+    throw new TypeError('Invalid config detected: "url" should be string!');
+  }
+  if (config.url.trim().length <= 0) {
+    throw new TypeError('Invalid config detected: "url" should not be empty!');
+  }
```

Comment:
suggest to include `undefined` checking here as well, need to run before 'string' checking.

Reply:
@curbengh Added in L9:

```js
  if (typeof config.url !== 'string') {
```

`undefined` just won't pass it.

Reply:
That check is inaccurate for `undefined` since it's an empty value.

Reply:
@curbengh I have already added an undefined test cases. It seems working properly.